pkg/gossiper/db.go

```diff
@@ -63,7 +64,31 @@ func (d *mydb) ListData() (int, []pair) {
 	iter.Release()
 	err := iter.Error()
 	if err != nil {
-		return DB_DELETE_ERROR, nil
+		return DB_DELETE_ERROR, nil, err
+	}
+	return SUCCEED, ans, nil
+}
+
+type batch struct {
+	leveldb.Batch
+}
+
+func (b *batch) Set(keyval []pair) {
+	for i := 0; i < len(keyval); i++ {
+		b.Put([]byte(keyval[i].key), []byte(keyval[i].val))
+	}
+}
+
+func (b *batch) Remove(keys []string) {
+	for i := 0; i < len(keys); i++ {
+		b.Delete([]byte(keys[i]))
+	}
+}
+
+func (d *mydb) Commit(mybatch *leveldb.Batch) (int, error) {
```

Comment:
don't need `mybatch *leveldb.Batch` as input parameter. you can just commit d.batch